server/src/main/java/org/elasticsearch/health/Diagnosis.java

```diff
@@ -136,21 +143,30 @@ public Collection<DiscoveryNode> getNodes() {
     public record Definition(String indicatorName, String id, String cause, String action, String helpURL) {}
 
     @Override
-    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-        builder.startObject();
-        builder.field("id", HEALTH_API_ID_PREFIX + definition.indicatorName + ":diagnosis:" + definition.id);
-        builder.field("cause", definition.cause);
-        builder.field("action", definition.action);
-
+    public Iterator<? extends ToXContent> toXContentChunked() {
+        Iterator<? extends ToXContent> resourcesIterator = Collections.emptyIterator();
         if (affectedResources != null && affectedResources.size() > 0) {
-            builder.startObject("affected_resources");
-            for (Resource affectedResource : affectedResources) {
-                affectedResource.toXContent(builder, params);
+            resourcesIterator = affectedResources.stream()
+                .flatMap(s -> StreamSupport.stream(Spliterators.spliteratorUnknownSize(s.toXContentChunked(), Spliterator.ORDERED), false))
+                .iterator();
+        }
```

Comment:
I think the previously asked `Iterators.concat` pattern question applies here as well?